boost_histogram/_internal/axis.py

```diff
@@ -0,0 +1,373 @@
+        if i < 0:
+            i += self._ax.size
+        if i >= self._ax.size:
+            raise IndexError("Out of range access")
+        return self.bin(i)
+
+    def __iter__(self):
+        return self._ax.__iter__()
+
+    @property
+    def edges(self):
+        return self._ax.edges
+
+    @property
+    def centers(self):
+        """
+        An array of bin centers.
+        """
+        return self._ax.centers
+
+    @property
+    def widths(self):
+        """
+        An array of bin widths.
+        """
+        return self._ax.widths
+
+
+class Regular(Axis):
+    __slots__ = ()
+    _CLASSES = {
+        ca.regular_uoflow,
+        ca.regular_uoflow_growth,
+        ca.regular_uflow,
+        ca.regular_oflow,
+        ca.regular_none,
+        ca.regular_log,
+        ca.regular_pow,
+        ca.regular_sqrt,
+        ca.regular_numpy,
+        ca.circular,
+    }
+
+    def __init__(self, bins, start, stop, **kwargs):
+        """
+        Make a regular axis with nice keyword arguments for underflow,
+        overflow, and growth.
+
+        Parameters
+        ----------
+        bins : int
+            The number of bins between start and stop
+        start : float
+            The beginning value for the axis
+        stop : float
+            The ending value for the axis
+        metadata : object
+            Any Python object to attach to the axis, like a label.
+        underflow : bool = True
+            Enable the underflow bin
+        overflow : bool = True
+            Enable the overflow bin
+        growth : bool = False
+            Allow the axis to grow if a value is encountered out of range.
+            Be careful, the axis will grow as large as needed.
+
+        See also
+        --------
+        .sqrt
+            Square root regular binning
+        .pow
+            Power binning
+        .log
+            Log space binning
+        .circular
+            Circular _to_axis
+        """
+
+        with KWArgs(kwargs) as k:
+            metadata = k.optional("metadata")
+            options = k.options(
+                underflow=True, overflow=True, growth=False, circular=False
+            )
+
+        if options == {"growth", "underflow", "overflow"}:
+            self._ax = ca.regular_uoflow_growth(bins, start, stop, metadata)
+        elif options == {"underflow", "overflow"}:
+            self._ax = ca.regular_uoflow(bins, start, stop, metadata)
+        elif options == {"underflow"}:
+            self._ax = ca.regular_uflow(bins, start, stop, metadata)
+        elif options == {"overflow"}:
+            self._ax = ca.regular_oflow(bins, start, stop, metadata)
+        elif options == {"circular", "underflow", "overflow"}:
+            self._ax = ca.circular(bins, start, stop, metadata)
+        elif options == set():
+            self._ax = ca.regular_none(bins, start, stop, metadata)
+        else:
+            raise KeyError("Unsupported collection of options")
+
+    @classmethod
```

Comment:
This is a change that slipped by me. I think we should not have these class methods, since people should use the `transform` keyword.